javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -1284,6 +1284,10 @@ class BigDecimal() extends Number with Comparable[BigDecimal] {
     }
   }
 
+  @inline override def __scala_==(other: _Object): scala.Boolean = {
+    this.equals(other)
+  }
```

Comment:
This should not be here. The codegen is supposed to automatically add this in the vtable because `BigDecimal` overrides `equals`, as explained by the comment at
https://github.com/scala-native/scala-native/blob/b49e74efe7079bdf415b5724da4acb87c5567210/nativelib/src/main/scala/java/lang/Object.scala#L39-L44
If that's not the case, that's a problem in the codegen; not in the library.

In general, user space classes should *never* implement `__scala_==`, since that is an implementation detail of Scala Native.

Reply:
Removed